### PR TITLE
TS Rewrite

### PR DESCRIPTION
**Pull request by ghostzero**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "@as207111/whois",
-  "version": "1.0.2",
+  "version": "2.0.0-beta",
```
Comment (ghostzero) [inline package.json:3]: Propper version tagging. TBD

File: src/lib/Client.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            Accept: "application/json",
+            Authorization: `Bearer ${this._apiKey}`,
+          },
+        }
+      );
+
+      if (response.status !== 200) {
+        switch (response.status) {
+          case 503:
+            throw new ResponseError(503, "Database Error");
```
Comment (ghostzero) [inline src/lib/Client.ts:32]: We should use `Service Unavailable`, because it's more generic.

File: src/lib/Client.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+          case 409:
+            throw new ResponseError(409, "Lookup Error");
```
Comment (ghostzero) [inline src/lib/Client.ts:35]: We should use `Conflict`, because it's more generic.

File: src/lib/Client.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+          method: "GET",
+          headers: {
+            Accept: "application/json",
+            Authorization: `Bearer ${this._apiKey}`,
+          },
+        }
+      );
+
+      if (response.status !== 200) {
+        switch (response.status) {
```
Comment (ghostzero) [inline src/lib/Client.ts:30]: Maybe we should also pass some details to the ResponseError, like the response.message, errors like 409, 503 have a message defined.

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,5 @@
 /node_modules
 /.idea
-package-lock.json
\ No newline at end of file
+package-lock.json
+
+dist/
```
Comment (ghostzero) [inline .gitignore:5]: no new-line at the end

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+        "sourceMap": true,
+        "outDir": "dist",
+        "strict": true,
+        "noImplicitAny": false,
+        "lib": [
+            "esnext"
+        ],
+        "esModuleInterop": true
+    }
+}
```
Comment (ghostzero) [inline tsconfig.json:12]: no new-line at the end

File: .gitignore (hunk 5 above)
Comment (ghostzero) [inline .gitignore:5]: I don't know right now if it's recommended or not to push the `dist` directory.

File: src/lib/Client.ts (hunk 2 above)
Comment (whopiyush) [reply src/lib/Client.ts:32]: the messages were taken from the data you sent on the discord server, I'll change it

File: src/lib/Client.ts (hunk 3 above)
Comment (whopiyush) [reply src/lib/Client.ts:35]: the messages were taken from the data you sent on the discord server, I'll change it

File: src/lib/Client.ts (hunk 4 above)
Comment (whopiyush) [reply src/lib/Client.ts:30]: the messages were taken from the data you sent on the discord server, if the statusCode is not in the switch case, it does send the response data. a more nice way would be to just pass the errors field. I'll look into it

File: .gitignore (hunk 5 above)
Comment (whopiyush) [reply .gitignore:5]: generally, we don't push the build folder

Comment (ghostzero) [reply .gitignore:5]: Okay ✅

**Pull request merged**